src/main/java/org/jboss/naming/remote/client/InitialContextFactory.java

```diff
@@ -61,6 +70,71 @@
      * @throws NamingException if an error occurs
      */
     public Context getInitialContext(final Hashtable<?, ?> environment) throws NamingException {
-        return delegate.getInitialContext(environment);
+        return delegate.getInitialContext(findAndCreateClientProperties(environment));
+    }
+
+    private static ClassLoader getClientClassLoader() {
+        final ClassLoader tccl = secureGetContextClassLoader();
+        if (tccl != null) {
+            return tccl;
+        }
+        return InitialContextFactory.class.getClassLoader();
+    }
+
+    private Properties findAndCreateClientProperties(final Hashtable<?, ?> env) {
+        // First load the props file if it exists
+        Properties props = findClientProperties();
+        if (props == null) {
+            props = new Properties();
+        }
+        // Now override with naming env entries
+        for (Map.Entry<?, ?> entry : env.entrySet()) {
+            if (entry.getKey() instanceof String && entry.getValue() instanceof String) {
+                props.setProperty((String) entry.getKey(), (String) entry.getValue());
+            }
+        }
+        return props;
+    }
+
+    private Properties findClientProperties() {
+        final ClassLoader classLoader = getClientClassLoader();
+        Messages.log.debug("Looking for " + CLIENT_PROPS_FILE_NAME + " using classloader " + classLoader);
+        // find from classloader
+        final InputStream clientPropsInputStream = classLoader.getResourceAsStream(CLIENT_PROPS_FILE_NAME);
+        if (clientPropsInputStream != null) {
+            Messages.log.debug("Found " + CLIENT_PROPS_FILE_NAME + " using classloader " + classLoader);
+            Messages.log.oldClientPropertyFileDeprecated();
+            final Properties clientProps = new Properties();
+            try {
+                clientProps.load(clientPropsInputStream);
```

Comment:
You should almost always load properties using the `load(Reader)` method.  Create the reader in a try-with-resources block to avoid closing in the `finally` block below.

Reply:
The reader should be an `InputStreamReader` using `StandardCharsets.UTF_8`.